sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -402,6 +425,109 @@ def _get_environment(
         env = self._environment_from_row(row)
         return env
 
+    @transactional()
+    def add_interval(
+        self,
+        snapshot: Snapshot,
+        start: TimeLike,
+        end: TimeLike,
+        is_dev: bool = False,
+    ) -> None:
+        logger.info("Adding interval for snapshot %s", snapshot.snapshot_id)
+
+        self.engine_adapter.insert_append(
+            self.intervals_table,
+            _intervals_to_df([snapshot], start, end, is_dev, False),
+            columns_to_types=self._interval_columns_to_types,
+        )
+
+    @transactional()
+    def remove_interval(
+        self,
+        snapshots: t.Iterable[SnapshotInfoLike],
+        start: TimeLike,
+        end: TimeLike,
+        all_snapshots: t.Optional[t.Iterable[Snapshot]] = None,
+    ) -> None:
+        all_snapshots = all_snapshots or self._get_snapshots_with_same_version(snapshots)
+
+        if logger.isEnabledFor(logging.INFO):
+            snapshot_ids = ", ".join([str(s.snapshot_id) for s in snapshots])
+            logger.info("Removing interval for snapshots: %s", snapshot_ids)
+
+        self.engine_adapter.insert_append(
+            self.intervals_table,
+            _intervals_to_df(all_snapshots, start, end, False, True),
+            columns_to_types=self._interval_columns_to_types,
+        )
+
+    @transactional()
+    def compact_intervals(self) -> None:
+        query = (
+            exp.select(
+                "id", "name", "identifier", "version", "start_ts", "end_ts", "is_dev", "is_removed"
+            )
+            .from_(self.intervals_table)
+            .join(
```

Comment:
what's the point of this join? why not just do a where statement on the from?

Reply:
Because I don't like the syntax around constructing an IN condition with a tuple.